### PR TITLE
[PECO-1016-2] Add handling for special types

### PR DESCRIPTION
**Pull request by nithinkdb**
In this PR, we add further handling to allow for special types to be set via the DBSQLParams variable.

### CODE REVIEW

File: parameters.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+const (
+	String SqlType = iota
+	Date
+	Timestamp
+	Float
+	Decimal
+	Double
+	Integer
+	Bigint
```
Comment (rcypher-databricks) [inline parameters.go:28]: can we change these to BigInt, SmallInt, and TinyInt

**Review by rcypher-databricks — COMMENTED** [review]
Let's add a new example showing how to run a parameterized query.
We should also add a test driver_e2e_test.go so that we have a test that exercises the interaction of our connection as a NamedValueChecker.

**Comment by nithinkdb** [comment]
> Let's add a new example showing how to run a parameterized query. We should also add a test driver_e2e_test.go so that we have a test that exercises the interaction of our connection as a NamedValueChecker.

Created this follow up work item to track both of these: https://databricks.atlassian.net/browse/PECO-1048

**Review by rcypher-databricks — APPROVED** [review]
Looks good.  I think we might want to change the names of the SqlType constants but we can talk about that for the next PR.

**Pull request merged**